File: 백준/Silver/10866. 덱/덱.py
from collections import deque
import sys
input = sys.stdin.readline

def command(x):
    if x[0] == 'push_front':
        arr.appendleft(x[1])
    elif x[0] == 'push_back':
        arr.append(x[1])
    elif x[0] == 'size':
        print(len(arr))
    elif x[0] == 'empty':
        if len(arr) == 0:
            print(1)
        else:
            print(0)
    elif len(arr) == 0:
        print(-1)
    else:
        if x[0] == 'pop_front':
            print(arr.popleft())
        elif x[0] == 'pop_back':
            print(arr.pop())
        elif x[0] == 'front':
            print(arr[0])
        elif x[0] == 'back':
            print(arr[-1])

n = int(input())
arr = deque([])
for _ in range(n):
    command(list(input().split()))

File: 백준/Silver/10866. 덱/덱.cc
#include <iostream>
#include <deque>
using namespace std;
int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);
  int N, num; cin >> N;
  string command;
  deque<int> dq;
  while (N--) {
    cin >> command;
    if (command == "push_front") {
      cin >> num;
      dq.push_front(num);
    } else if (command == "push_back") {
      cin >> num;
      dq.push_back(num);
    } else if (command == "pop_front") {
      if (dq.empty()) cout << "-1\n";
      else {
        cout << dq.front() << '\n';
        dq.pop_front();
      }
    } else if (command == "pop_back") {
      if (dq.empty()) cout << "-1\n";
      else {
        cout << dq.back() << '\n';
        dq.pop_back();
      }
    } else if (command == "size") cout << dq.size() << '\n';
    else if (command == "empty") cout << (dq.empty() ? 1 : 0) << '\n';
    else if (command == "front") cout << (dq.empty() ? -1 : dq.front()) << '\n';
    else if (command == "back") cout << (dq.empty() ? -1 : dq.back()) << '\n';
  }
  return 0;
}